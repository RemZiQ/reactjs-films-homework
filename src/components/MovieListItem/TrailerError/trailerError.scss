.trailerError_container {
  position: absolute;
  z-index: 100;
  width: 500px;
  height: 200px;
  background-color: rgba(0, 0, 0, 0.9);
  text-align: center;
  line-height: 200px;
  color: white;
  font-size: 26px;
}
