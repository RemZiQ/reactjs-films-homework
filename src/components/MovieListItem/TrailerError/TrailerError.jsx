import React, { Component } from 'react';
import propTypes from 'prop-types';

import './trailerError.scss';

class TrailerError extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error) {
    this.setState({ hasError: error });
  }

  render() {
    const { hasError } = this.state;
    const { children } = this.props;
    if (hasError) {
      return (
        <div className="trailerError_container">
          <h2>Sorry, we dont have trailer for that film.</h2>
        </div>
      );
    }

    return children;
  }
}


TrailerError.propTypes = {
  children: propTypes.node.isRequired,
};

export default TrailerError;
