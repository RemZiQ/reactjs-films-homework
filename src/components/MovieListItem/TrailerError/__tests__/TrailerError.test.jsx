import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { TrailerError } from '../TrailerError';


it('TrailerError renders well', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<TrailerError />);
  expect(result).toMatchSnapshot();
});
