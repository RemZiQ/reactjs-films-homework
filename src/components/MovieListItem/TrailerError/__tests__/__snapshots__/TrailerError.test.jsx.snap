// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailerError renders well 1`] = `<div />`;
